### ISSUE TITLE
script to reduce the number of features in model

### ISSUE THREAD

**Issue by msdw** [dmlc/xgboost#658]
Hello

thank you for xgboost, it is an amazing implementation of GBM.
I look for a way to reduce the number of variables used in xgboost without reducing too much (less than 1 point of auc) the loss in my performance criterion (auc)
My train dataset contains 20,000 features. xgboost retains in model nearly 3,000 features. I would like to iterate so select only the top 50 variables in order to deploy the model on a large volume (several million). How to do that ? Do you have a script to reduce the number of  variables included in the model?

Thanks for your help.

**Comment by pommedeterresautee** [comment]
The easiest way to do it is to leverage the feature importance feature (both in R and Python).
There is no automatic script to do it.

If I answered your question, please close the issue.

Kind regards,
Michael

**Comment by msdw** [comment]
Hi Michael,

Thanks for your return.
Just to share, I develop a loop in with I keep only top 95% features based on feature importance and run it until model has less 30 features.
Anyway thanks.
